### ISSUE TITLE
Task that depends on multiple `wall_clock` xtriggers will submit when first is satisfied

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#5783]
### Reproducible Example

Weird example but demonstrates the bug

```ini
[scheduler]
    allow implicit tasks = True
[scheduling]
    initial cycle point = 2018-01-01
    [[xtriggers]]
        clock_1 = wall_clock()  # satisfied when now > 2018-01-01
        clock_2 = wall_clock(offset=P10Y)  # satisfied when now > 2028-01-01
    [[graph]]
        R1 = @clock_1 & @clock_2 => foo
```

```
INFO - [20180101T0000Z/foo waiting(runahead) job:00 flows:1] => waiting
INFO - xtrigger satisfied: clock_1 = wall_clock(trigger_time=1514764800)
INFO - xtrigger satisfied: clock_2 = wall_clock(trigger_time=1514764800) <-- note same trigger time logged
...
INFO - [20180101T0000Z/foo preparing job:01 flows:1] => submitted
```

### Expected Behaviour

`foo` only submits when all the `wall_clock` xtriggers it depends on are satisfied.

### Additional Context

I think the log is wrong - I don't think `clock_2` is satisfied yet; it's just ignored by `foo`. If you change it to
```py
R1 = """
   @clock_1 & clock_2 => foo
   @clock_2 => bar
"""
```
then `foo` still submits immediately but `bar` does not submit yet

**Comment by wxtim** [comment]
This is caused by `itask.get_clock_trigger_time` caching the clock trigger time, causing it to return the first value it was ever fed every time.

**Comment by oliver-sanders** [comment]
Closed by #5791
